recipe/meta.yaml

```diff
@@ -46,6 +46,10 @@ outputs:
 
     requirements:
       build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - cython                                 # [build_platform != target_platform]
+        - ninja                                  # [build_platform != target_platform]
```

Comment:
```suggestion
        - cmake                                  # [build_platform != target_platform]
        - ninja                                  # [build_platform != target_platform]
```

Reply:
good call - also looks like we need scipy here, last run it failed w/ attempting to compile it from source

```
    Build type: native build
    Project name: SciPy
    Project version: 1.11.1

    ../../meson.build:1:0: ERROR: Could not invoke sanity test executable: [Errno 86] Bad CPU type in executable: '/private/tmp/pip-install-7zs395if/scipy_593f7293e9d24cb09e2113e3b60838ee/.mesonpy-e19sq8zj/build/meson-private/sanitycheckc.exe'.

    A full log can be found at /private/tmp/pip-install-7zs395if/scipy_593f7293e9d24cb09e2113e3b60838ee/.mesonpy-e19sq8zj/build/meson-logs/meson-log.txt
    error: subprocess-exited-with-error

    × Preparing metadata (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
```

Reply:
hmm - that didn't work. seems like the build is called w/

```
2023-07-24T21:31:54.2259160Z   note: This error originates from a subprocess, and is likely not a problem with pip.
2023-07-24T21:31:54.2268420Z   full command: /Users/runner/miniforge3/conda-bld/implicit_1690233977768/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/python /Users/runner/miniforge3/conda-bld/implicit_1690233977768/_build_env/lib/python3.10/site-packages/pip/__pip-runner__.py install --ignore-installed --no-user --prefix /private/tmp/pip-build-env-r74a8zt3/overlay --no-warn-script-location -v --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=42' wheel 'scikit-build>=0.13.1' 'Cython>=0.24' 'scipy>=0.16'
```
and scipy is still attempting to be built from source, possibly because of the  `--ignore-installed` flag.

I wonder if we need the ` --no-build-isolation` flag here

Reply:
using `--no-build-isolation` fixed the OSX ARM builds (and broke everything else).  at least its some progress =)